part2/countries/src/App.js

```diff
@@ -0,0 +1,122 @@
+
+import React, { useState, useEffect } from 'react'
+import axios from 'axios'
+
+const Filter = ({ handleFilterCountry }) => (
+  <>
+    <p> Find countries </p>
+    <input onChange={handleFilterCountry} />
+
+  </>
+)
+
+const Language = ({ language }) => (
+  <li>{language}{language.key}</li>
+)
+
+
+const Country = ({ country }) => {
+  return (
+    <>
+      <h3>{country.name}</h3>
+      <p>Capital: {country.capital}</p>
+      <p>Population: {country.population}</p>
+      <h3>Language(s)</h3>
+
+      <ul>
+        {country.languages.map((language) =>
+          <Language key={language.name} language={language.name} />
+        )}
+      </ul>
+
+      <img alt="flag" src={country.flag} />
+
+
+    </>
+  )
+}
+
+const List = ({ countriesList }) => {
+
+// relationship between first two conditions - 
+// they somewhat cooporate, but I don't really appreciate inconsistency
+// both should react (appear and disappear) according to the state of input field 
+// not to array length. If input is empty - "feel free", 
+// otherwise "too many matches" || countries list of 10 || the single country
+
+  if (countriesList.length > 20) {
+    return (
+      <p>Feel free to search for a country</p>
+    )
+  }
+
+  else if (countriesList.length > 10) {
+    return (
+      <p>Too many matches, specify another filter</p>
+    )
+
+  } else if (countriesList.length === 1) {
+    return (
+      <>
+        <ul>
+          {countriesList.map((country) =>
+            <Country key={country.name} country={country} />
+          )}
+        </ul>
```

Comment:
Še tas `<ul>` šķiet nav vajadzīgs. Jo `Country` nerenderē `<li>`... Not to mention, ka šeit notiek operācija ar vienu valsti un `ul` translates to `unordered list`, bet viena valsts nav gluži saraksts, ne?